src/classdef.cpp

```diff
@@ -3324,7 +3324,7 @@ bool ClassDefImpl::isLinkable() const
   }
   else
   {
-    return isReference() || isLinkableInProject();
+    return (isReference() && !isInValidTagReference()) || isLinkableInProject();
```

Comment:
try to avoid double negation, i.e. `isValidTagReference()` is easier to read and understand than `!isInvalidTagReference()` , is there any case where `isReference()` does not also imply `isValidTagReference()`? If not then better make it part of `isReference()`

Reply:
A quick search gave that there were 8 places where the new construct is used and 131 more constructs where only `isReference` is used.

Regarding the double negation: Yes it is most of the times hard to understand what is going on in that case though here I found it much clearer and easier to use the double negation.

Reply:
The followup question I have is why is the valid tag check needed in the 8 cases, but not in the other 131 cases? What is the criteria to include/not include the extra check? I ask from a maintenance point of view, i.e. if I add a new `isReference()` somewhere when do I also need the valid tag check?

Reply:
This is always hard to tell, the basic idea is to prevent to create a link that points to nowhere.
The `!isInValidReference` has to be added where a request is made to check / create a link, as this cannot be done due to the missing filename.